apps/web/scripts/kernel/handlers/connect.ts

```diff
@@ -29,16 +32,45 @@ export const handleConnect: MessengerCallback = async (payload, channel) => {
 		}
 	}
 
-	const keyResponse = await modules.storage.find(selectors.allKeys);
-	const publicKeys = keyResponse.docs as PublicKeyDocument[];
-	console.log('All public keys: ', publicKeys);
+	const popup = await handleOpenPopup(
+		PopupType.REQUEST_CONNECT_POPUP,
+		requestId,
+	);
 
-	console.log('send back response:', publicKeys);
-	channel.postMessage({
-		from: 'walless@kernel',
-		requestId: payload.requestId,
-		publicKeys: publicKeys,
-	});
+	requestMap[requestId] = {
+		channel,
+		payload,
+		popup,
+		resolve: false,
+	};
+};
+
+export const handleResolveConnect: MessengerCallback = async (payload) => {
+	const { requestId = '' } = payload;
+	const requestSource = requestMap[requestId];
+	if (!requestSource?.resolve) {
+		if (!payload.isApproved) {
+			requestSource.channel.postMessage({
+				from: 'walless@kernel',
+				requestId,
+				message: Message.REJECT_REQUEST_CONNECT,
+			});
```

Comment:
I mean we need to create a convention for messaging about success and failed

Reply:
I also think we need fields like `error` and `responseCode`. Is that what you mean?

Reply:
I mean in messaging, when receiving a reject, It will raise error. We need an util to handle this (inside messaging engine code) before sdk receive response. @fluctlight-kayaba We need your opinion.

Reply:
I have no clear clue if Messaging should/could handle that, initial idea is to decouple child modules, it should cover thing that generic enough. While error handing may different between different parts (e.g sdk vs content-script), keep in mind that sdk don't use messing module at all.

While we need unified convention/util to handle success/failure, maybe it won't be part of the messaging package, I don't know, @tanlethanh if you already have direction/idea just go  ahead.